astro/tests/integration_test.go

```diff
@@ -52,6 +60,75 @@ func getSessionDirs(sessionBaseDir string) ([]string, error) {
 	return sessionDirs, nil
 }
 
+// compiles the astro binary and returns the path to it.
+func compileAstro(dir string) (string, error) {
+	astroPath := filepath.Join(dir, "astro")
+	packageName := "github.com/uber/astro/astro/cli/astro"
+	out, err := exec.Command("go", "build", "-o", astroPath, packageName).CombinedOutput()
+	if err != nil {
+		return "", errors.New(string(out))
+	}
+	return astroPath, nil
+}
+
+func TestPlanInterrupted(t *testing.T) {
+	fakeTerraformPath := "fixtures/terraform"
+	require.True(t, utils.FileExists(fakeTerraformPath))
+	fakeTerraformDir, err := filepath.Abs(filepath.Dir(fakeTerraformPath))
+	require.NoError(t, err)
+
+	oldPath := os.Getenv("PATH")
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", fakeTerraformDir, oldPath))
+	defer os.Setenv("PATH", oldPath)
+
+	tmpdir, err := ioutil.TempDir("", "astro-binary")
+	defer os.RemoveAll(tmpdir)
+	require.NoError(t, err)
+
+	astroBinary, err := compileAstro(tmpdir)
+	require.NoError(t, err)
+	command := exec.Command(astroBinary, "plan")
+
+	fixtureAbsPath, err := filepath.Abs("fixtures/plan-interrupted")
+	require.NoError(t, err)
+	command.Dir = fixtureAbsPath
+
+	stdoutBytes := &bytes.Buffer{}
+	stderrBytes := &bytes.Buffer{}
+	command.Stdout = stdoutBytes
+	command.Stderr = stderrBytes
+
+	var cmdErr error
+	processChan := make(chan struct{}, 1)
+	go func() {
+		defer close(processChan)
+		cmdErr = command.Run()
+		processChan <- struct{}{}
+	}()
+
+	// let astro start terraform processes
+	time.Sleep(1000 * time.Millisecond)
+	require.NoError(t, command.Process.Signal(os.Interrupt))
+
+	select {
+	case <-processChan:
+	case <-time.After(5 * time.Second):
+		// force kill the process after timeout
+		require.NoError(t, command.Process.Signal(syscall.SIGKILL))
+	}
+
+	require.Error(t, cmdErr)
+	require.Equal(t, 1, command.ProcessState.ExitCode())
+
+	stdout := stdoutBytes.String()
+	stderr := stderrBytes.String()
+	assert.Contains(t, stdout, "\nReceived signal: interrupt, cancelling operation...\n")
+	assert.Regexp(t, `foo\d{2}:.*ERROR`, stderr)
+	assert.NotRegexp(t, `foo\d{2}:`, stdout)
+	assert.NotRegexp(t, `bar\d{2}:`, stdout)
+	assert.NotRegexp(t, `bar\d{2}:`, stderr)
```

Comment:
Why can we make an assumption that `bar` won't be executed?

Reply:
This was the goal of this test: to be sure that no tests are executed after we sent an interrupt signal to astro process. As astro uses at most 10 parallel executions at a time, barXX should not be executed until all fooXX are finished (there are exactly 10 of them).